extensions/integration-tests/src/test/schemaCompare.test.ts

```diff
@@ -270,6 +274,231 @@ suite('Schema compare integration test suite @DacFx@', () => {
 		const excludeResult3 = await schemaCompareService.schemaCompareIncludeExcludeNode(operationId, t2Difference, false, azdata.TaskExecutionMode.execute);
 		assertIncludeExcludeResult(excludeResult3, true, 0, 0);
 	});
+
+	test('Cancel schema compare dacpac to dacpac comparison', async function () {
+		this.timeout(5 * 60 * 1000);
+		assert(schemaCompareService, 'Schema Compare Service Provider is not available');
+		const now = new Date();
+		const operationId = 'testOperationId_' + now.getTime().toString();
+
+		const source: mssql.SchemaCompareEndpointInfo = {
+			endpointType: mssql.SchemaCompareEndpointType.Dacpac,
+			packageFilePath: dacpac1,
+			serverDisplayName: '',
+			serverName: '',
+			databaseName: '',
+			ownerUri: '',
+			connectionDetails: undefined
+		};
+		const target: mssql.SchemaCompareEndpointInfo = {
+			endpointType: mssql.SchemaCompareEndpointType.Dacpac,
+			packageFilePath: dacpac2,
+			serverDisplayName: '',
+			serverName: '',
+			databaseName: '',
+			ownerUri: '',
+			connectionDetails: undefined
+		};
+
+		const promiseCompare = schemaCompareService.schemaCompare(operationId, source, target, azdata.TaskExecutionMode.execute, null);
+		const promiseCancel = schemaCompareService.schemaCompareCancel(operationId);
+
+		const schemaCompareResult = await promiseCompare;
+		const schemaCompareCancelResult = await promiseCancel;
+
+		assert(schemaCompareResult.success === false, `Expected: failure in schema compare. Actual: Success`);
+		assert(schemaCompareCancelResult.success === true, `Expected: success in schema compare cancel. Actual: Failure`);
+	});
+
+	test('Schema compare dacpac to database comparison with publishing some changes and then compare again', async function () {
+		this.timeout(5 * 60 * 1000);
+		let server = await getStandaloneServer();
+		const ownerUri = await getConnectionUri(server);
+		const now = new Date();
+		const operationId = 'testOperationId_' + now.getTime().toString();
+		const targetDB: string = 'ads_schemaCompare_targetDB_' + now.getTime().toString();
+
+		try {
+			assert(dacfxService, 'DacFx Service Provider is not available');
+			let result = await dacfxService.deployDacpac(dacpac1, targetDB, true, ownerUri, azdata.TaskExecutionMode.execute);
```

Comment:
would it be better to run tsql to create the target database and then run a script to create some objects so that this test doesn't depend on the dacfxService?

Reply:
@udeeshagautam , @llali : Any thoughts on this?

Reply:
Reasoning behind running script vs deploy service: the benefit of creating the db separately is that if something breaks in DacFx or in the future it takes different arguments, the schema compare tests won't be affected. 
It might also be a little faster since the query is directly being executed instead of having to go through all the DacFx logic.

Reply:
I updated the new tests to create the database and add objects separately, instead of using dacFxService for deploying a dacpac.

Reply:
I wasn't able to figure out the "ownerUri" issue that was popping up in lab runs (and not in local testing). I have created this workitem for now and will work on it later: https://github.com/microsoft/azuredatastudio/issues/14647.

Reply:
agree with Kim that we should not use DacFx to create test dbs 

---
In reply to: [586683061](https://github.com/microsoft/azuredatastudio/pull/14479#discussion_r586683061) [](ancestors = 586683061)